src/UI/Settings/SettingsLayout.js

```diff
@@ -248,5 +252,24 @@ module.exports = Marionette.Layout.extend({
         } else {
             $('body').removeClass('show-advanced-settings');
         }
+    },
+    
+    _setProxySettingsState : function() {
+        var value = Config.getValue(Config.Keys.ProxySettings);
+
+        if (value !== 'off') {
+            $('body').addClass('show-proxy-settings');
+        }
+    },
+    
+    _toggleProxySettings : function() {
+        console.log("Proxy Toggled")
+        var value = $('.x-proxy-settings').val();
+        Config.setValue(Config.Keys.ProxySettings, value);
+        if (value === 'off') {
+            $('body').removeClass('show-proxy-settings');
```

Comment:
I don't think you actually need to all the way up in the root of the tree, you could just add it to the current view. `this.$el.removeClass`